### PR TITLE
cli: Enable multi-tenant demo by default

### PR DESCRIPTION
**Pull request by ajstorm**
#71026 added support for cockroach demo to run in multi-tenant mode. In an
effort to get broader testing around multi-tenancy, this PR enables
multi-tenant mode by default.

This is currently a WIP as we first need to see what breaks in CI with this
enabled.

Release note (cli change): cockroach demo now runs by default in multi-tenant
mode. See #71026 for more details.

### CODE REVIEW

**Review by knz — APPROVED** [review]

**Comment by ajstorm** [comment]
Removed the DNM tag, as CI is now passing.  Ready for another look.

**Review by knz — APPROVED** [review]
still LGTM, with nit

> Reviewed 9 of 9 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-:-MnRthkr4I3W4qySGXjy:b-701ndo)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @rafiss)

___
*[pkg/cli/demo.go, line 201 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-MnRtNvH76TfhRUZ8pGx:-MnRtNvH76TfhRUZ8pGy:bpcz66w) ([raw file](https://github.com/cockroachdb/cockroach/blob/198d2a323634f6b64047d3e46f3d663e421e0864/pkg/cli/demo.go#L201)):*
> ```
> flagSetForCmd(cmd)
> ```

please factor the computation of flagSetForCmd(cmd) to a common var above in this function and use it in both places where it's useful.
___
*[pkg/cli/interactive_tests/test_demo_workload.tcl, line 42 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-MnRtaWnCNwC7rHUslNz:-MnRtaWnCNwC7rHUslO-:b-9ecat4) ([raw file](https://github.com/cockroachdb/cockroach/blob/198d2a323634f6b64047d3e46f3d663e421e0864/pkg/cli/interactive_tests/test_demo_workload.tcl#L42)):*
> ```Tcl
> # Reset the timeout.
> set timeout 30
> # Need to disable multi-tenant mode here, as splitting is not supported.
> ```

maybe link the relevant issue in a comment?
___
*[pkg/cli/interactive_tests/test_disable_replication.tcl, line 10 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-MnRtd00EYQNZQbQoRsA:-MnRtd00EYQNZQbQoRsB:bvrwsev) ([raw file](https://github.com/cockroachdb/cockroach/blob/198d2a323634f6b64047d3e46f3d663e421e0864/pkg/cli/interactive_tests/test_disable_replication.tcl#L10)):*
> ```Tcl
> 
> start_test "Check that demo disables replication properly"
> # Disable multitenant until zone configs are properly enabled in tenants.
> ```

maybe link the relevant issue in a comment

**Review by ajstorm — COMMENTED** [review]
Thanks for the second review.  I'm going to hold off on borsing this until the morning so that I can draft a communication to give the organization the heads-up.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-:-MnSZkWp0sDAq7-E7SEv:b9iiw8y)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @rafiss)

___
*[pkg/cli/demo.go, line 201 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-MnRtNvH76TfhRUZ8pGx:-MnSXam9EacWlB7K_G4w:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/198d2a323634f6b64047d3e46f3d663e421e0864/pkg/cli/demo.go#L201)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

> ```
> flagSetForCmd(cmd)
> ```

please factor the computation of flagSetForCmd(cmd) to a common var above in this function and use it in both places where it's useful.
</blockquote></details>

Done.
___
*[pkg/cli/interactive_tests/test_demo_workload.tcl, line 42 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-MnRtaWnCNwC7rHUslNz:-MnSYJq51f60vNq7yP5s:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/198d2a323634f6b64047d3e46f3d663e421e0864/pkg/cli/interactive_tests/test_demo_workload.tcl#L42)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

maybe link the relevant issue in a comment?
</blockquote></details>

Done.
___
*[pkg/cli/interactive_tests/test_disable_replication.tcl, line 10 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/71988#-MnRtd00EYQNZQbQoRsA:-MnSZ9OG-dhLOSts8FXG:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/198d2a323634f6b64047d3e46f3d663e421e0864/pkg/cli/interactive_tests/test_disable_replication.tcl#L10)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

maybe link the relevant issue in a comment
</blockquote></details>

Done.

**Comment by ajstorm** [comment]
TFTR!

bors r=knz

**Pull request merged**
